subxt/src/backend/unstable/follow_stream_driver.rs

```diff
@@ -379,6 +380,77 @@ struct SubscriberDetails<Hash: BlockHash> {
     waker: Option<Waker>,
 }
 
+pin_project! {
+    /// A stream that subscribes to finalized blocks
+    /// and indicates whether a block was missed if the connection was lost.
+    #[derive(Debug)]
+    pub struct FollowStreamFinalizedHeads<Hash: BlockHash> {
+        #[pin]
+        stream: FollowStreamDriverSubscription<Hash>,
+        sub_id: Option<String>,
+        last_seen_block: Option<BlockRef<Hash>>,
+    }
+}
+
+impl<Hash: BlockHash> FollowStreamFinalizedHeads<Hash> {
+    pub fn new(stream: FollowStreamDriverSubscription<Hash>) -> Self {
+        Self {
+            stream,
+            sub_id: None,
+            last_seen_block: None,
+        }
+    }
+}
+
+impl<Hash: BlockHash> Stream for FollowStreamFinalizedHeads<Hash> {
```

Comment:
This looks good to me!As a nit I'd be tempted to put it in a different file like the stream in `storage_items.rs` is (or put some of this stuff into a utils file or something) since it's not really a part of the follow stream driver code

Reply:
yepp, the thingy was that the FollowStreamSubscription wasn't exported outside this and I placed here to avoid sharing it with `pub crate`.

I can move it if it ok to make it pub crate :)